RPC operations can throw UndeclaredThrowableException in some circumstances.

When an RPC client operation times out, Artemis throws an `ActiveMQConnectionTimedOutException` exception:
```java
2017-05-16T15:47:01,203 [WARN ] NodeTerminalView - Cash balance RPC failed: null
java.lang.reflect.UndeclaredThrowableException: null
        at com.sun.proxy.$Proxy27.getCashBalances(Unknown Source) ~[?:?]
        at net.corda.demobench.views.NodeTerminalView.pollCashBalances(NodeTerminalView.kt:217) [demobench-0.12-SNAPSHOT.jar:?]
        ...

Caused by: org.apache.activemq.artemis.api.core.ActiveMQConnectionTimedOutException: AMQ119014: Timed out after waiting 30,000 ms for response when sending packet 43
        at org.apache.activemq.artemis.core.protocol.core.impl.ChannelImpl.sendBlocking(ChannelImpl.java:399) ~[artemis-core-client-1.5.3.jar:1.5.3]
        at org.apache.activemq.artemis.core.protocol.core.impl.ChannelImpl.sendBlocking(ChannelImpl.java:318) ~[artemis-core-client-1.5.3.jar:1.5.3]
        at org.apache.activemq.artemis.core.protocol.core.impl.ActiveMQSessionContext.simpleCommit(ActiveMQSessionContext.java:328) ~[artemis-core-client-1.5.3.jar:1.5.3]
        at org.apache.activemq.artemis.core.client.impl.ClientSessionImpl.commit(ClientSessionImpl.java:505) ~[artemis-core-client-1.5.3.jar:1.5.3]
        at net.corda.client.rpc.internal.RPCClientProxyHandler.invoke(RPCClientProxyHandler.kt:209) ~[rpc-0.12-SNAPSHOT.jar:?]
        ... 8 more

```
However, this is a checked exception and none of the methods on `CordaRPCOps` declare it as thrown. This causes interoperability problems with Java, triggering an `UndeclaredThrowableException`.

I can reproduce this problem as follows:
- Launch DemoBench and create a simple notary node.
- Once the notary node has booted, suspend it using `kill -stop <pid>` (MacOS, Linux)
- Check `$HOME/demobench/demobench.log` for DemoBench's RPC client to notice that the node has stopped responding.

We need to wrap these checked Artemis exceptions inside unchecked exceptions.

--- a/client/rpc/src/main/kotlin/net/corda/client/rpc/internal/RPCClientProxyHandler.kt
+++ b/client/rpc/src/main/kotlin/net/corda/client/rpc/internal/RPCClientProxyHandler.kt
@@ -22,7 +22,6 @@ import org.apache.activemq.artemis.api.core.SimpleString
 import org.apache.activemq.artemis.api.core.client.ActiveMQClient.DEFAULT_ACK_BATCH_SIZE
 import org.apache.activemq.artemis.api.core.client.ClientMessage
 import org.apache.activemq.artemis.api.core.client.ServerLocator
-import org.apache.activemq.artemis.core.client.impl.ClientConsumerInternal
 import rx.Notification
 import rx.Observable
 import rx.subjects.UnicastSubject
@@ -77,7 +76,7 @@ class RPCClientProxyHandler(
         val log = loggerFor<RPCClientProxyHandler>()
         // Note that this KryoPool is not yet capable of deserialising Observables, it requires Proxy-specific context
         // to do that. However it may still be used for serialisation of RPC requests and related messages.
-        val kryoPool = KryoPool.Builder { RPCKryo(RpcClientObservableSerializer) }.build()
+        val kryoPool: KryoPool = KryoPool.Builder { RPCKryo(RpcClientObservableSerializer) }.build()
         // To check whether toString() is being invoked
         val toStringMethod: Method = Object::toString.javaMethod!!
     }
@@ -209,6 +208,12 @@ class RPCClientProxyHandler(
                 it.session.commit()
             }
             return replyFuture.getOrThrow()
+        } catch (e: RuntimeException) {
+            // Already an unchecked exception, so just rethrow it
+            throw e
+        } catch (e: Exception) {
+            // This must be a checked exception, so wrap it
+            throw RPCException(e.message ?: "", e)
         } finally {
             callSiteMap?.remove(rpcId.toLong)
         }

